Object types naming rules

stefanorosanelli opened the pull request:
This PR introduces some addtional rules on object types naming:

- a list of reserved names is provided - `plugins/BEdita/Core/config/reserved.php` now very short, expected to grow - that cannot be used as objcet type names
- this list may be used in other use cases - like property names
- plural `name` and `singular` must be different - since plural `name` form is the preferred one as a consequence we may use `media_item` and `news_item` as singular form for `media` and `news`

Differentiating `name` and `singular` enables, among other things, unambiguous different root types in *graphql* for singular and plural forms.
 
Code:
- `NotReserved` and `ObjectTypesValidator` class introduced

stefanorosanelli committed:
feat: add `config/reserved.php` array of reserved names + validation rules

stefanorosanelli committed:
chore: improve coverage and make code sniffer happy again

stefanorosanelli committed:
chore: add reserved names

stefanorosanelli committed:
chore: make scrutinizer happy

batopa commented on plugins/BEdita/API/tests/TestCase/Controller/ObjectTypesControllerTest.php:
```diff
@@ -584,7 +585,34 @@ public function testAddInvalid()
+        $this->assertContentType('application/vnd.api+json');
+        $this->assertEquals($count, TableRegistry::get('ObjectTypes')->find()->count());
 
+        // add reserved word failrure
```
typo: failure

batopa commented on plugins/BEdita/Core/src/Model/Validation/ObjectTypesValidator.php:
```diff
@@ -0,0 +1,83 @@
+
+                    return ($context['data']['name'] !== $value);
+                },
+                'message' => 'Name and singular fields must be different'
```
Should we use `__d('bedita', 'Name and singular fields must be different')` for string translations?

batopa commented on plugins/BEdita/Core/tests/TestCase/Model/Validation/NotReservedTest.php:
```diff
@@ -0,0 +1,67 @@
+     * Test reserved rule.
+     *
+     * @param string|null $expected Expected alias, or `null`.
+     * @param string $alias Alias to search for.
```
the `@param`s are wrong

stefanorosanelli committed:
chore: small typos and doc changes

fquffio commented on plugins/BEdita/API/tests/TestCase/Controller/ObjectTypesControllerTest.php:
```diff
@@ -584,7 +585,34 @@ public function testAddInvalid()
+        $this->assertContentType('application/vnd.api+json');
+        $this->assertEquals($count, TableRegistry::get('ObjectTypes')->find()->count());
 
+        // add reserved word failure
```
I'd prefer to have this test split into its own test case, to better distinguish possible failures.

scrutinizer-auto-fixer committed:
Scrutinizer Auto-Fixes

This commit consists of patches automatically generated for this project on https://scrutinizer-ci.com

stefanorosanelli committed:
Merge pull request #1360 from bedita/scrutinizer-patch-1

Scrutinizer Auto-Fixes

stefanorosanelli committed:
tests: split cases in different methods

stefanorosanelli committed:
Merge branch 'feature/v4/object-types-naming' of github.com:bedita/bedita into feature/v4/object-types-naming

stefanorosanelli committed:
feat: migration for `news_item` and `media_item`

stefanorosanelli committed:
test: update fixtures w `news_item` and `media_item`

stefanorosanelli committed:
Merge branch '4-cactus' into feature/v4/object-types-naming
